spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -1421,6 +1435,7 @@ public void testPartitionsTable() {
     }
   }
 
+  @SuppressWarnings("MethodLength")
```

Comment:
`testPartitionsTableLastUpdatedSnapshot` has more than 150 lines due to the new assertions I added.  Please let me know if they make sense to you, thanks.

Reply:
I think we can have some helper methods.  I'm ok to do it in a follow up pr, maybe @dramaticlly can also help as the original author ?

Reply:
yeah I also run into this problem before. If your datafile total size assertion is already covered in `testPartitionsTable` then I dont think it's necessary to ido it again in `testPartitionsTableLastUpdatedSnapshot` since what you tested are mostly the same scenario. 

Also if you want to move file assertion into some helper method to reduce line count, that's probably more preferred than suppress warning. 

I can also help with follow up PR if needed, it's up to you

Reply:
I was thinking we can make a helper for making the expected?  But not sure if we can do it so that  spotless wont blow it back up to one field per line

Reply:
Also some other things like, example:

```
    Assert.assertEquals(
        "rewrite replaced 2 manifests",
        2,
        Iterables.size(rewriteManifestResult.rewrittenManifests()));
    Assert.assertEquals(
        "rewrite added 1 manifests", 1, Iterables.size(rewriteManifestResult.addedManifests()));
```

can be

```
assertResult(rwriteManifestResult, 2, 1)

Reply:
@dramaticlly @szehon-ho Thanks for your feedback.

Extracted those assertions to a helper method, see https://github.com/apache/iceberg/pull/7920/commits/0d423c219739a1e26d167d7a2bd176a2cb516544